arch/Kconfig

```diff
@@ -149,7 +149,7 @@ config ARCH_IS_SET
 
 menu "General Architecture Options"
 
-source "$(ARCH_DIR)/common/Kconfig"
+source "arch/common/Kconfig"
```

Comment:
I think you need to keep this for downstream users that want to override it no? In my local change I set `${ARCH_DIR}`  in unittest.cmake to `${ZEPHYR_BASE}/arch` if it wasn't already set

Reply:
> I think you need to keep this for downstream users that want to override it no? 

The `source "$(ARCH_DIR)/common/Kconfig"` was introduced here #44358 in commit 070422db46af53cd692f01320b410d3ba07e57df but should never have been introduced in the first place.

This change foces every user with an out-of-tree arch to implement a `common/Kconfig` file, even if they don't need it.
If users wants to create a common Kconfig for all their arch and source such file, they can do that themselves, Zephyr shouldn't require such file to be present.

>  In my local change I set `${ARCH_DIR}` in unittest.cmake to `${ZEPHYR_BASE}/arch` if it wasn't already set

and you can still do that, and your arch Kconfig will be picked up from here:
https://github.com/zephyrproject-rtos/zephyr/blob/ccc0315e14341879cc466156a4c925dd27b06525/arch/Kconfig#L12
for your custom arch.